### ISSUE TITLE
[Xbase][Refactoring] Renaming Inferred private methods is not working

### ISSUE THREAD

**Issue by cdietrich** [eclipse/xtext-eclipse#76]
Renaming Inferred private methods is not working

take the domainmodel example and adapt the inferrer
```
			members += entity.toMethod("get"+entity.name, Void.TYPE.typeRef) [
				body = ''''''
				visibility = JvmVisibility.PRIVATE
			]
```

and refactor `Demo` in the following model:

```
package abc {
	entity Demo {
	}
}
```

gives

`Refactoring is not applicable`

this is cause by the `RenameVirtualMethodProcessor` created for the method which does not like private methods (see `MethodChecks.isVirtual`)

i asume we should change

`JdtRenameRefactoringProcessorFactory.createRenameProcessor`

to something like

```
case IJavaElement.METHOD:
					if(((IMethod)element).isConstructor()) 
						break;
					if (Flags.isStatic(((IMethod) element).getFlags()) || Flags.isPrivate(((IMethod) element).getFlags()))
						return new RenameNonVirtualMethodProcessor((IMethod) element);
					else
						return new RenameVirtualMethodProcessor((IMethod) element);
```

@JanKoehnlein what do you think

**Comment by JanKoehnlein** [comment]
Sounds good. Does it break tests?

In general we lack a communication channel to describe how the name of a model element is transformed during inference. The current implementation relies on the fact that the inferred elements keep their URI fragments.

**Comment by cdietrich** [comment]
i will give it a try (with our tests and the ones from the customer as well)

**Comment by cdietrich** [comment]
problem is reproducable in xtend as well

**Comment by cdietrich** [comment]
similar problem for enum literals

**Comment by cdietrich** [comment]
Fixed in Xtext 2.11
